dataland-frontend/src/components/forms/CreateEuTaxonomyForNonFinancials.vue

```diff
@@ -0,0 +1,343 @@
+<template>
+  <Card class="col-12 page-wrapper-card p-3">
+    <template #title
+      ><span data-test="pageWrapperTitle">
+        {{ editMode ? "Edit" : "Create" }} EU Taxonomy Dataset for a Non-Financial Company/Service</span
+      ></template
+    >
+    <template #content>
+      <div v-if="waitingForData" class="d-center-div text-center px-7 py-4">
+        <p class="font-medium text-xl">Loading Eu Taxonomy For Non Financials data...</p>
+        <em class="pi pi-spinner pi-spin" aria-hidden="true" style="z-index: 20; color: #e67f3f" />
+      </div>
+      <div v-else class="grid uploadFormWrapper">
+        <div id="uploadForm" class="text-left uploadForm col-9">
+          <FormKit
+            v-model="companyAssociatedEuTaxonomyDataForNonFinancials"
+            :actions="false"
+            type="form"
+            :id="formId"
+            :name="formId"
+            @submit="postEuTaxonomyForNonFinancialsData"
+            @submit-invalid="checkCustomInputs"
+          >
+            <FormKit type="hidden" name="companyId" :model-value="companyID" disabled="true" />
+            <div class="uploadFormSection grid">
+              <div class="col-3 p-3 topicLabel">
+                <h4 id="reportingPeriod" class="anchor title">Reporting Period</h4>
+              </div>
+              <div class="col-9 form-field formFields uploaded-files">
+                <UploadFormHeader
+                  :label="'Reporting Period'"
+                  :description="'The reporting period the dataset belongs to (e.g. a fiscal year).'"
+                  :is-required="true"
+                />
+                <div class="lg:col-4 md:col-6 col-12 pl-0">
+                  <Calendar
+                    data-test="reportingPeriod"
+                    v-model="reportingPeriod"
+                    inputId="icon"
+                    :showIcon="true"
+                    view="year"
+                    dateFormat="yy"
+                    validation="required"
+                  />
+                </div>
+
+                <FormKit type="hidden" :modelValue="reportingPeriodYear" name="reportingPeriod" />
+              </div>
+            </div>
+
+            <FormKit type="group" name="data" label="data">
+              <div
+                v-for="category in euTaxonomyForNonFinancialsDataModel"
+                :key="category"
+                :label="category.label"
+                :name="category.name"
+              >
+                <div class="uploadFormSection grid" v-for="subcategory in category.subcategories" :key="subcategory">
```

Comment:
We are using similar form-construction loops all across Dataland. Investigate if it is possible / easy to consolidate this loop into a component.

Reply:
This might be a seperate refactoring task.

Reply:
Story created.